common/src/main/java/ch/admin/bag/covidcertificate/common/qr/QRCodeReaderHelper.kt

```diff
@@ -52,36 +55,45 @@ object QRCodeReaderHelper {
 		return decoded
 	}
 
-	fun pdfToBitmap(context: Context, pdfFile: File): ArrayList<Bitmap> {
+	fun pdfToBitmaps(context: Context, pdfFile: File): ArrayList<Bitmap> {
 		val bitmaps = arrayListOf<Bitmap>()
 		try {
 			val renderer = PdfRenderer(ParcelFileDescriptor.open(pdfFile, ParcelFileDescriptor.MODE_READ_ONLY))
-			var bitmap: Bitmap?
 			val pageCount = renderer.pageCount
 
-			for (i in 0 until min(pageCount, 5)) {
+			for (i in 0 until min(pageCount, PDF_PAGE_LIMIT)) {
 				val page: PdfRenderer.Page = renderer.openPage(i)
-				val width = context.resources.displayMetrics.densityDpi / 72 * page.width
-				val height = context.resources.displayMetrics.densityDpi / 72 * page.height
-				bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
 
-				val canvas = Canvas(bitmap)
-				canvas.drawColor(Color.WHITE)
-				canvas.drawBitmap(bitmap, 0.0f, 0.0f, null)
+				// PDF width/height are given in "points pt" such that 1 pt = 1/72 inch
+				// => 72 "dots-per-inch dpi" <==> scale = 1
+				val scale: Float = context.resources.displayMetrics.densityDpi / 72f
+				val pixelWidth = (scale * page.width).roundToInt()
+				val pixelHeight = (scale * page.height).roundToInt()
 
-				val r = Rect(0, 0, width, height)
-				page.render(bitmap, r, null, PdfRenderer.Page.RENDER_MODE_FOR_PRINT)
-				bitmaps.add(bitmap)
+				// Try both the scaled and the unscaled version
+				bitmaps.add(page.renderToBitmap(pixelWidth, pixelHeight))
+				bitmaps.add(page.renderToBitmap(page.width, page.height))
 
 				page.close()
 			}
 			renderer.close()
-
 		} catch (ex: Exception) {
 			ex.printStackTrace()
 		}
 
 		return bitmaps
 	}
+
+	private fun PdfRenderer.Page.renderToBitmap(pixelWidth: Int, pixelHeight: Int): Bitmap {
+		val bitmap = Bitmap.createBitmap(pixelWidth, pixelHeight, Bitmap.Config.ARGB_8888)
+
+		// Make sure the bitmap's background is not transparent (which can cause issues for QR code detection)
+		Canvas(bitmap).drawColor(Color.WHITE)
```

Comment:
I think this could also be achieved with `.eraseColor(Color.WHITE)` on the Bitmap directly instead of creating a new temporary Canvas object